docs/api.md

```diff
@@ -0,0 +1,1683 @@
+  useEffect(() => fcl.currentUser().subscribe(setUser), []); // sets the callback for FCL to use
+
+  if (user.loggedIn) {
+    return (
+      <div>
+        <span>{user?.addr ?? "No Address"}</span>
+        <button onClick={fcl.unauthenticate}>Log Out</button> {/* once logged out in setUser(user) will be called */}
+      </div>
+    );
+  } else {
+    return (
+      <div>
+        <button onClick={fcl.logIn}>Log In</button>{" "}
+        {/* once logged in setUser(user) will be called */}
+        <button onClick={fcl.signUp}>Sign Up</button> {/* once signed up, setUser(user) will be called */}
+      </div>
+    );
+  }
+}
+```
+
+---
+
+## `fcl.currentUser().snapshot()`
+
+Returns the [current user](#`CurrentUserObject`) object. This is the same object that is set and available on [`fcl.currentUser().subscribe(callback)`](<##`fcl.currentUser().subscribe(callback)`>).
+
+### Usage
+
+```javascript
+// returns the current user object
+const user = fcl.currentUser().snapshot();
+
+// subscribes to the current user object and logs to console on changes
+fcl.currentUser().subscribe(console.log);
+```
+
+---
+
+## `fcl.currentUser().authenticate()`
+
+Equivalent to `fcl.authenticate()`.
+
+---
+
+## `fcl.currentUser().unauthenticate()`
+
+Equivalent to `fcl.unauthenticate()`.
+
+---
+
+## `fcl.currentUser().authorization()`
+
+Equivalent to `fcl.authz`.
+
+---
+
+# On-chain Interactions
+
+> :loudspeaker: **These methods can be used in browsers and NodeJS.**
+
+These methods allows dapps to interact directly with the Flow blockchain via a set of functions that currently use the [Access Node API](https://docs.onflow.org/access-api/).
+
+## Methods
+
+---
+
+## Query and Mutate Flow with Cadence
+
+If you want to run arbitrary Cadence scripts on the blockchain, these methods offer a convenient way to do so **without having to build, send, and decode interactions**.
+
+## `fcl.query({...options})`
+
+Allows you to submit scripts to query the blockchain.
+
+### Options
+
+_Pass in the following as a single object with the following keys.All keys are optional unless otherwise stated._
+
+| Key       | Type                                    | Description                                                                                                 |
+| --------- | --------------------------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `cadence` | string **(required)**                   | A valid cadence script.                                                                                     |
+| `args`    | [ArgumentFunction](#`ArgumentFunction`) | Any arguments to the script if needed should be supplied via a function that returns an array of arguments. |
+| `limit`   | number                                  | Compute limit for query. :tomato: WHAT UNITS ARE THESE IN?                                                  |
```

Comment:
```suggestion
| `limit`   | number                                  | Compute limit for query. Got to the [documentation](https://docs.onflow.org/flow-go-sdk/building-transactions/#gas-limit) for information about how Gas is calculated on Flow.                                          |
```

Reply:
Can we put a reccomended value for testing -> Maybe the same number used in kitty-items?

Reply:
This number is covered in the docs.

Reply:
There is a spelling mistake in the suggested update: `Got to the [documentation]` -> `Go to the [documentation]`